addons/stock_account/models/account_invoice.py

```diff
@@ -77,6 +77,56 @@ def _anglo_saxon_sale_move_lines(self, i_line):
                 ]
         return []
 
+    def _get_related_stock_moves(self):
+        """ To be overridden for customer invoices and vendor bills in order to
+        return the stock moves related to this invoice.
+        """
+        return self.env['stock.move']
+
+    def _get_products_set(self):
+        """ Returns a recordset of the products contained in this invoice's lines
+        """
+        return self.mapped('invoice_line_ids.product_id')
+
+    def _get_anglosaxon_interim_account(self, product):
+        """ Returns the interim account used in anglosaxon accounting for
+        this invoice"""
+        if self.type in ('out_invoice', 'out_refund'):
+            return product.product_tmpl_id._get_product_accounts()['stock_output']
+        elif self.type in ('in_invoice', 'in_refund'):
+            return product.product_tmpl_id.get_product_accounts()['stock_input']
+
+        return None
+
+    def invoice_validate(self):
+        super(AccountInvoice, self).invoice_validate()
+        self.anglo_saxon_reconcile_valuation(correct_invoice_interim_account=True)
```

Comment:
We have come through this approach, and we, at vauxoo, have hit a wall with performance issues 
Because having invoices with several Lines make the process to lengthy, Customers have complained about
having this reconciliation at validation step, thus we had to take another approach by means of scheduled actions at the end of the day, is a least active approach but causes less troubles to costumers.

Could you on your side try with invoices having at least 500 lines?

If you are willing to leave this feature as is at least give us, integrators, the ability to skip this feature, with a CheckOption in the company, thus allowing us to implement it in a passive way.

https://github.com/Vauxoo/addons-vauxoo/blob/8.0/account_anglo_saxon_stock_move_purchase/data/cron_job.xmlhttps://github.com/Vauxoo/addons-vauxoo/blob/8.0/account_anglo_saxon_stock_move_purchase/data/action_server_data.xml

https://github.com/Vauxoo/addons-vauxoo/blob/8.0/account_anglo_saxon_stock_move_sale/data/cron_job.xml
https://github.com/Vauxoo/addons-vauxoo/blob/8.0/account_anglo_saxon_stock_move_sale/data/action_server_data.xml

Reply:
let's try not to do any pre-emptive optimization. We've got a branch optimizing the reconciliation methods for better performances when called on several lines... So it might be already enough